File: Exp1-Visual_Surface_Detection.cpp
#include <graphics.h>
#include <bits/stdc++.h>

using namespace std;

void drawTriangle() {
    vector<int> x = {10, 50, 100};
    vector<int> y = {100, 20, 100};

    setcolor(GREEN);
    line(x[0], y[0], x[1], y[1]);
    line(x[1], y[1], x[2], y[2]);
    line(x[2], y[2], x[0], y[0]);

    setfillstyle(SOLID_FILL, GREEN);
    floodfill((x[0]+x[1]+x[2])/3, (y[0]+y[1]+y[2])/3, GREEN); 
}


void drawCircle()
{
    setcolor(BLUE);
    circle(100, 100, 45);
    setfillstyle(SOLID_FILL, BLUE);
    floodfill(101, 101, BLUE);
}

void drawRectangle()
{
    setcolor(RED);
    rectangle(100, 100, 180, 180);
    setfillstyle(SOLID_FILL, RED);
    floodfill(101, 101, RED);
}

int main()
{
    int gd = DETECT, gm = DETECT;
    initgraph(&gd, &gm, "");

    string sequence = "RCT";
    for (char x: sequence)
    {
        if (x == 'C') drawCircle();
        else if (x == 'T') drawTriangle();
        else drawRectangle();
    }

    getch();
    closegraph();
    return 0;
}

File: Exp1-Visual_Surface_Detection.py
import turtle

# --- Setup ---
screen = turtle.Screen()
screen.bgcolor("white")
screen.title("Visual Surface Detection")

pen = turtle.Turtle()
pen.speed(1)

# -----------------------
# Draw Triangle (same coords)
# -----------------------
def drawTriangle():
    x = [10, 50, 100]
    y = [100, 20, 100]

    pen.penup()
    pen.goto(x[0], y[0])
    pen.pendown()
    pen.fillcolor("green")
    pen.begin_fill()
    pen.goto(x[1], y[1])
    pen.goto(x[2], y[2])
    pen.goto(x[0], y[0])
    pen.end_fill()

# -----------------------
# Draw Circle
# -----------------------
def drawCircle():
    pen.penup()
    pen.goto(100, 100 - 45)  
    pen.pendown()
    pen.fillcolor("blue")
    pen.begin_fill()
    pen.circle(45)
    pen.end_fill()

# -----------------------
# Draw Rectangle
# -----------------------
def drawRectangle():
    x1, y1 = 100, 100
    x2, y2 = 180, 180

    pen.penup()
    pen.goto(x1, y1)
    pen.pendown()
    pen.fillcolor("red")
    pen.begin_fill()
    pen.goto(x2, y1)
    pen.goto(x2, y2)
    pen.goto(x1, y2)
    pen.goto(x1, y1)
    pen.end_fill()

# -----------------------
# Sequence of drawing
# -----------------------
sequence = "RCT"
for shape in sequence:
    if shape == "C":
        drawCircle()
    elif shape == "T":
        drawTriangle()
    else:
        drawRectangle()

turtle.done()
